Flappy Bird Clone/Assets/Scripts/Obstacles/ObstacleBehaviour.cs

```diff
@@ -0,0 +1,36 @@
+using System;
+using Player;
+using UnityEngine;
+
+namespace Obstacles
+{
+    public class ObstacleBehaviour : MonoBehaviour
+    {
+        // DELEGATES
+        public delegate void PlayerTouchedObstacle();
+
+        // EVENTS
+        public static event PlayerTouchedObstacle OnPlayerTouchedObstacle;
+        
+        // VARIABLES
+        private BoxCollider2D _collider;
+        
+        // METHODS
+        private void Awake()
+        {
+            _collider = GetComponent<BoxCollider2D>();
+        }
+
+        // EVENTS
+        private void OnCollisionEnter2D(Collision2D other)
+        {
+            if (_collider.isTrigger || !other.gameObject.TryGetComponent(out PlayerBehaviour player))
+            {
+                return;
+            }
+
+            OnPlayerTouchedObstacle?.Invoke();
```

Comment:
Why call this static instead callind a method in the object?

Reply:
Don't have a specific answer for this question, I was just trying different stuff and thought it could be ok to be static